### PR TITLE
Singleton skip

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: test/patterns.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+from vsc.utils.patterns import Singleton
+
+
+class ABC(object):
+    __metaclass__ = Singleton
+
+    def __init__(self, value):
+        """
+        Singleton metaclass with init that has args is probably the most stupid thing to have,
```
Comment (itkovian) [inline test/patterns.py]: I do not grok this comment. Why not be able to set stuff?

Comment (stdweird) [reply test/patterns.py]: check the tests, tell me that it makes sense

File: test/patterns.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+        abc = ABC(1)
+        self.assertEqual(abc.value, 1, "init with value 1")
+
+        abc2 = ABC(2)
+        self.assertEqual(abc2.value, 1,
+                         "2nd init with value 2, gives value from (first) singleton")
+        self.assertEqual(id(abc2), id(abc), "2nd instance is identical to 1st")
+
+        abc3 = ABC(2, skip_singleton=True)
```
Comment (itkovian) [inline test/patterns.py]: Wait wut? :) So not a singleton after all.

File: test/sandbox/testpkg/__init__.py [hunk 3]
```diff
@@ -1,4 +1,29 @@
...
+# vsc-base is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public License
+# along with vsc-base. If not, see <http://www.gnu.org/licenses/>.
+#
+#
 # Copyright 2016-2017 Ghent University
```
Comment (itkovian) [inline test/sandbox/testpkg/__init__.py]: Remove the below lines, they are double?

File: test/patterns.py (hunk 2 above)
Comment (stdweird) [reply test/patterns.py]: yes, so we can transition away from it in eg `VSC()`

**Comment by stdweird** [comment]
unittests fail, will have alook in a few days

File: test/patterns.py (hunk 1 above)
Comment (JensTimmerman) [reply test/patterns.py]: @itkovian if 2 things need the singleton to be initialised in a different way, it can't and shouldn't be a singleton?

**Comment by JensTimmerman** [comment]
this is a very ugly hack

**Pull request closed without merging**